src/patch-code-path-analysis.coffee

```diff
@@ -1,19 +1,46 @@
 CodePathAnalyzer = require './code-path-analysis/code-path-analyzer'
 
-module.exports = ->
-  try
-    ESLintCodePathAnalyzer = require(
-      'eslint/lib/code-path-analysis/code-path-analyzer'
-    )
-  catch
+PROGRAM_NODE_KEY = '__coffee__'
+
+monkeypatchMethod = (ESLintCodePathAnalyzer, isCurrentFileCoffeescript, key) ->
+  original = ESLintCodePathAnalyzer::[key]
+  dynamicallyDelegatingMonkeypatch = (...args) ->
+    if isCurrentFileCoffeescript.current
+      # eslint-disable-next-line coffee/no-invalid-this
+      CodePathAnalyzer::[key].apply @, args
+    else
+      # eslint-disable-next-line coffee/no-invalid-this
+      original.apply @, args
+  if key is 'enterNode'
+    ESLintCodePathAnalyzer::[key] = (...args) ->
```

Comment:
There wasn't an existing "lifecycle hook" as far as I could tell that could be used for applying + unapplying the monkeypatch per each linted file. So instead, I tag the top-level `Program` AST node with a special property indicating "this is Coffeescript" and then have the monkeypatched methods either dynamically route to the overridden implementation if we're currently linting a Coffeescript file (ie the last `Program` node we saw had the special property set) or their original implementation if not